### DISCUSSION TITLE
Does FXGL could load 3D model generated by JCSG ?

### DISCUSSION THREAD

**Discussion by chengenzhao** [AlmasB/FXGL#1245] [Q&A] [answered]
Hi:

I tried to load the 3D model by using assetLoader.load("sample-color.obj")
which is created by this project:
https://github.com/miho/JCSG/blob/master/src/main/java/eu/mihosoft/jcsg/Main.java
it generates two files: sample-color.obj and sample-color.mtl
I could use Xcode to open these 2 files and it works OK
but FXGL doesn't load it correctly
so is there any thing missing?

![截屏2023-03-14 21 23 56](https://user-images.githubusercontent.com/5525436/225014566-31697a3a-b73a-4cdf-adbc-14a1ee9f4234.png)

![截屏2023-03-14 21 25 32](https://user-images.githubusercontent.com/5525436/225015008-7c0da1af-a586-427c-9fa5-e089d769497a.png)

Thanks

**Comment by chengenzhao** [comment]
Seems like JCSG already have a obj importer in this directory:
https://github.com/miho/JCSG/tree/master/src/main/java/eu/mihosoft/jcsg/ext/openjfx/importers/obj

**Comment by AlmasB** [comment, marked as answer]
It looks like the mtl file didn't load correctly. Can you attach all the 3D model files here please, I'll take a look.

**Reply by chengenzhao** [reply]
> It looks like the mtl file didn't load correctly. Can you attach all the 3D model files here please, I'll take a look.

Sure the model file is attached
please take a look

[sample-color.zip](https://github.com/AlmasB/FXGL/files/10984852/sample-color.zip)

**Reply by AlmasB** [reply]
I've just fixed one issue related to incorrectly loading `.obj` when no vertex normals are specified. This may have fixed the above issue. Please check when you can.

**Reply by chengenzhao** [reply]
it works
thx a lot
